python/taichi/_kernels.py

```diff
@@ -19,9 +19,10 @@
 
 # A set of helper (meta)functions
 @kernel
-def fill_tensor(tensor: template(), val: template()):
-    for I in grouped(tensor):
-        tensor[I] = val
+def fill_field(field: template(), val: template()):
+    value = ops.cast(val, field.dtype)
```

Comment:
Have a question here: does this `cast` operation introduces extra instructions when `val` and `field` are of the same type?